mitylib/report.py

```diff
@@ -1,962 +1,475 @@
-            # annotations
-            consequence_idx = vep_headers.index('Consequence')
-            
-            # get the index of the annotations that don't contain the word stream
-            not_stream = ['stream' not in x[consequence_idx] for x in VEP]
-            stream_idx = [i for i, x in enumerate(not_stream) if x]
-            # print(stream_idx)
-            
-            # remove the sublists that have 'stream'in the consequence from 
-            # the VEP annotations
-            VEP = [VEP[i] for i in stream_idx]
-            # print(VEP)
-            # remove empty sublists
-            VEP = [x for x in VEP if x != []]
-            # print(VEP)
-            
-            # if there are no annotations left, completely remove the sublist
-            if VEP == []:
-                # fill with dot
-                VEP = [['.'] * len(vep_headers)]
-            else:
-                VEP = [[element or '.' for element in sublist] for sublist in
-                       VEP]
-            
-            # print(VEP)
-            
-            # get all the consequences for each variant in a single list
-            # so that we can see what the highest impact is
-            csq = [x[consequence_idx] for x in VEP]
-            # some of them are split by '&'
-            csq = [x.split("&") for x in csq]
-            # unlist
-            csq = sum(csq, [])
-            # print(csq)
-            
-            # print(impact)
-            if csq == ['.']:
-                # impact = ["."]
-                highest_impact = "."
+
+class Vep:
+    """
+    Provides methods to handle vepped files.
+    """
+
+    def __init__(self, vcf_obj):
+        self.vcf_obj = vcf_obj
+        self.vepped = self.check_vepped()
+
+        if self.vepped:
+            self.vep_keys = self.generate_vep_keys()
+            self.vep_excel_headers = self.generate_vep_excel_headers()
+
+    def get_vep_name(self, name):
+        """
+        Capitalises vep name, replaces _ with <space> and adds "VEP" to the end.
+        """
+        name = name.upper()
+        name = name.replace("_", " ")
+        name = name + " VEP"
+
+        return name
+
+    def generate_vep_excel_headers(self):
+        """
+        Generates vep excel headers based on the vep keys.
+        """
+        vep_excel_headers = ["HIGHEST IMPACT VEP"]
+        for name in self.vep_keys:
+            vep_excel_headers.append(self.get_vep_name(name))
+
+        return vep_excel_headers
+
+    def find_highest_impact(self, impacts):
+        """
+        Returns the highest impact in the list of consequences. Raises error if
+        none of the consequences match.
+        """
+        if "HIGH" in impacts:
+            return "HIGH"
+        if "MODERATE" in impacts:
+            return "MODERATE"
+        if "MODIFIER" in impacts:
+            return "MODIFIER"
+        if "LOW" in impacts:
+            return "LOW"
+
+        raise ValueError("Unknown SO term.")
```

Comment:
I wonder if it is the best behaviour to raise an error if there is no matching impact. Could we perhaps just leave that field blank in the report and issue a warning? Just thinking about instance where this happens on a large VCF for one of the last variants of the file.. presumably the user has waited a while for an output before the error happened.